[16.0][FIX] beesdoo_product_label: change logo field from binary to image

mihien opened the pull request:
By default in 16 a binary field is stored as an attachment. So, after a migration, binary data that used to be in the model's table in previous versions will not be accessible anymore as the ORM will look them up in ir.attachment instead.

huguesdk reviewed (changes requested):
good catch! i would approve like this, but the question is now: why do we want to store it in the database table instead of as an `ir.attachment`? maybe this was not intentional. this was done in #236.

after checking in odoo's code the use cases for `attachment=False`, it is only for computed files (like a qr code). so this probably needs a migration script instead of this change.

edit: moreover, it needs to become a `fields.Image` field instead of `fields.Binary`.

huguesdk committed:
[FIX] change logo field from binary to image

change the logo field from binary to image and add a migration script to
convert values from the database during the migration to 16.0 (as binary
fields are stored by default in a column in a database in odoo 12.0, but
use attachments by default from odoo 13.0).

huguesdk commented:
i converted the field to a `fields.Image` and added a migration script for `16.0.1.0.0`, so that it will only run during migrations to 16.0, but not for new databases in 16.0 that already installed the module.

huguesdk commented:
/ocabot merge patch
